Mathlib/LinearAlgebra/Span.lean

```diff
@@ -583,6 +583,17 @@ theorem apply_mem_span_image_of_mem_span [RingHomSurjective σ₁₂] (f : M →
   exact Submodule.mem_map_of_mem h
 #align submodule.apply_mem_span_image_of_mem_span Submodule.apply_mem_span_image_of_mem_span
 
+theorem mem_span_of_apply_mem_span_image [RingHomSurjective σ₁₂] {f : M →ₛₗ[σ₁₂] M₂} {x : M}
+    {s : Set M} (hf : Function.Injective f) (h : f x ∈ Submodule.span R₂ (f '' s)) :
+    x ∈ Submodule.span R s := by
+  rwa [← Submodule.mem_comap, ← Submodule.map_span, Submodule.comap_map_eq_of_injective hf] at h
```

Comment:
Doesn't this work direclty for the `iff` version? If it does, I would say we can keep only the `iff` version if that works for you.